packages/xo-server/src/xo-mixins/pool.js

```diff
@@ -0,0 +1,96 @@
+import { difference, flatten, isEmpty, uniq } from 'lodash'
+
+export default class Pools {
+  constructor(xo) {
+    this._xo = xo
+  }
+
+  async mergeInto({ sources: sourceIds, target, force }) {
+    const { _xo } = this
+    const targetHost = _xo.getObject(target.master)
+    const sources = []
+    const sourcePatches = {}
+
+    // Check hosts compatibility.
+    for (const sourceId of sourceIds) {
+      const source = _xo.getObject(sourceId)
+      const sourceHost = _xo.getObject(source.master)
+      if (sourceHost.productBrand !== targetHost.productBrand) {
```

Comment:
Shouldn't we check the version as well?

Reply:
@olivierlambert Is it possible to join a XS host in a pool with a different version?

Reply:
I'm not sure. I would say no but ask someone (@benjamreis if he's around) to make a quick test to check this.

Reply:
That's impossible. Here's the return of  `xe pool-join` when attempting to join a 7.6 to a 8.0 master and the other way around: `The hosts in this pool are not compatible.`

Reply:
@benjamreis Thanks!

@Rajaa-BARHTAOUI IMHO it makes sense to check this before trying to patch the hosts.

Reply:
@julien-f  I'm just waiting for test: `join 7.5 to 7.6`.

Reply:
You could have merged both tests but that's fine.